rotkehlchen/premium/premium.py

```diff
@@ -333,27 +417,48 @@ def _register_new_device(self, device_id: str) -> None:
         - PremiumAuthenticationError: if the queried API returns a 401 error
         """
         log.debug(f'Registering new device {device_id}')
+        if (
+            (system_platform := platform.system()) == 'Linux' and
+            (docker_container := check_docker_container()) is not None
+        ):  # in the case of docker where we always run under linux, get the container id
+            system_platform = DOCKER_PLATFORM_KEY
+            device_name = docker_container
+        else:
+            device_name = str(uuid4())  # can be edited later.
+
         try:
             response = self.session.put(
                 url=f'{self.rotki_nest}{(method := "devices")}',
                 json=self.sign(
                     method=method,
                     device_identifier=device_id,
-                    device_name=str(uuid4()),  # can be edited later.
-                    platform=platform.system(),
+                    device_name=device_name,
+                    platform=system_platform,
                 ),
                 timeout=ROTKEHLCHEN_SERVER_TIMEOUT,
             )
         except requests.exceptions.RequestException as e:
             raise RemoteError(f'Failed to register device due to: {e}') from e
 
         if response.status_code in (HTTPStatus.CREATED, HTTPStatus.CONFLICT):
+            # Cache Docker device info on successful registration
+            if system_platform == DOCKER_PLATFORM_KEY:
+                self._set_docker_device_info(device_id, get_system_spec()['rotkehlchen'])
```

Comment:
why did you resovle this? The same problems here. InvalidVersion and KeyError can be raised, no? This would kill the greenlet.

Reply:
mentioned that keyerror means that our function changed and this wasn't updated and changed get_system_spec to handle the possible errors raised. Defaults to version 0.0.0